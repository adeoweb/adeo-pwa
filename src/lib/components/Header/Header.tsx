import React, { Suspense } from 'react';
import { Link, Route } from 'src/lib/drivers';
import Logo from 'src/lib/components/Logo';
import TopBar from './TopBar';
import DesktopMenu from 'src/lib/components/Header/DesktopMenu';
import { useHeader } from 'src/peregrine/lib/talons/adeoweb/Header/useHeader';
import HeaderContact from 'src/lib/components/Header/HeaderContact';
import CREATE_CART_MUTATION from '../../queries/createCart.graphql';
import GET_CART_DETAILS_QUERY from '../../queries/getCartDetails.graphql';
import MiniCart from 'src/lib/components/Header/MiniCart';
import { useCartTrigger } from 'src/peregrine/lib/talons/adeoweb/Header/useCartTrigger';
import { useNavigationTrigger } from 'src/peregrine/lib/talons/adeoweb/Header/useNavigationTrigger';

const SearchBar = React.lazy(() => import('src/lib/components/SearchBar'));

const Header = () => {
    const { handleSearchTriggerClick, searchOpen } = useHeader();
    const { handleOpenNavigation } = useNavigationTrigger();

    // TODO: redo search bar placeholder?
    const searchBarFallback = <div>Loading...</div>;

    const searchBar = (
        <Suspense fallback={searchBarFallback}>
            <Route
                render={({ history, location }) => (
                    <SearchBar
                        handleTriggerClick={handleSearchTriggerClick}
                        isOpen={searchOpen}
                        history={history}
                        location={location}
                    />
                )}
            />
        </Suspense>
    );

    useCartTrigger({
        createCartMutation: CREATE_CART_MUTATION,
        getCartDetailsQuery: GET_CART_DETAILS_QUERY
    });

    return (
        <header className="header">
            <TopBar />
            <div className="header-middle">
                <div className="container">
                    <div className="header-left">
                        <Link to={'/'} className="logo">
                            <Logo />
                        </Link>
                    </div>

                    <div className="header-center">{searchBar}</div>

                    <div className="header-right">
                        <button
                            className="mobile-menu-toggler"
                            type="button"
                            onClick={handleOpenNavigation}
                        >
                            <i className="icon-menu" />
                        </button>
                        <HeaderContact />
                        <MiniCart />
                    </div>
                </div>
            </div>
            <div className="sticky-wrapper">
                <div className="header-bottom sticky-header">
                    <div className="container">
                        <Link to={'/'} className="logo">
                            <Logo />
                        </Link>
                        <div className="dropdown cart-dropdown">
                            <a
                                href="/"
                                className="dropdown-toggle"
                                role="button"
                                data-toggle="dropdown"
                                aria-haspopup="true"
                                aria-expanded="false"
                                data-display="static"
                            >
                                <span className="cart-count">2</span>
                            </a>

                            <div className="dropdown-menu">
                                <div className="dropdownmenu-wrapper">
                                    <div className="dropdown-cart-header">
                                        <span>2 Items</span>

                                        <a href="cart.html">View Cart</a>
                                    </div>
                                    <div className="dropdown-cart-products">
                                        <div className="product">
                                            <div className="product-details">
                                                <h4 className="product-title">
                                                    <a href="product.html">
                                                        Woman Ring
                                                    </a>
                                                </h4>

                                                <span className="cart-product-info">
                                                    <span className="cart-product-qty">
                                                        1
                                                    </span>
                                                    x $99.00
                                                </span>
                                            </div>
                                            <figure className="product-image-container">
                                                <a
                                                    href="product.html"
                                                    className="product-image"
                                                >
                                                    <img
                                                        src="src/lib/assets/images/products/cart/product-1.jpg"
                                                        alt="product"
                                                    />
                                                </a>
                                                <a
                                                    href="/"
                                                    className="btn-remove"
                                                    title="Remove Product"
                                                >
                                                    <i className="icon-cancel" />
                                                </a>
                                            </figure>
                                        </div>
                                        <div className="product">
                                            <div className="product-details">
                                                <h4 className="product-title">
                                                    <a href="product.html">
                                                        Woman Necklace
                                                    </a>
                                                </h4>

                                                <span className="cart-product-info">
                                                    <span className="cart-product-qty">
                                                        1
                                                    </span>
                                                    x $35.00
                                                </span>
                                            </div>
                                            <figure className="product-image-container">
                                                <a
                                                    href="product.html"
                                                    className="product-image"
                                                >
                                                    <img
                                                        src="src/lib/assets/images/products/cart/product-2.jpg"
                                                        alt="product"
                                                    />
                                                </a>
                                                <a
                                                    href="/"
                                                    className="btn-remove"
                                                    title="Remove Product"
                                                >
                                                    <i className="icon-cancel" />
                                                </a>
                                            </figure>
                                        </div>
                                    </div>
                                    <div className="dropdown-cart-total">
                                        <span>Total</span>
                                        <span className="cart-total-price">
                                            $134.00
                                        </span>
                                    </div>
                                    <div className="dropdown-cart-action">
                                        <a
                                            href="checkout-shipping.html"
                                            className="btn btn-block"
                                        >
                                            Checkout
                                        </a>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <DesktopMenu />
                    </div>
                </div>
            </div>
        </header>
    );
};

export default Header;
