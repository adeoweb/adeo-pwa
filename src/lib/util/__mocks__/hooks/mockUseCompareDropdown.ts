import { TUseCompareDropdown } from 'src/peregrine/lib/talons/adeoweb/Product/useCompareDropdown';

const compareDropdownTestRef = { current: null };

const mock: TUseCompareDropdown = {
    compareDropdownOpen: false,
    openCompareDropdown: () => {},
    compareDropdownRef: compareDropdownTestRef
};

export default mock;
