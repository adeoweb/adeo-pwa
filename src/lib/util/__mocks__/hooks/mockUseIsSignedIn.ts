import { TUseSignIn } from 'src/peregrine/lib/talons/adeoweb/IsSignedIn/useIsSignedIn';

const mock: TUseSignIn = { isSignedIn: false };

export default mock;
