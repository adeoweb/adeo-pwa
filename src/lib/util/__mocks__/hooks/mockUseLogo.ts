import { TUseLogo } from 'src/peregrine/lib/talons/adeoweb/Logo/useLogo';

const mock: TUseLogo = {
    storeName: 'test',
    logoSrc: 'test',
    logoAlt: 'test',
    logoHeight: 100,
    logoWidth: 100
};

export default mock;
