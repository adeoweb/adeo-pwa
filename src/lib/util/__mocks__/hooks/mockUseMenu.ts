import { TUseMenu } from 'src/peregrine/lib/talons/adeoweb/Menu/useMenu';

const mock: TUseMenu = {
    error: null,
    children: [
        {
            children_count: '2',
            id: 10,
            include_in_menu: 1,
            name: 'Mock name',
            position: 2,
            url_path: 'mockurl.html',
            children: [
                {
                    children_count: '2',
                    id: 11,
                    include_in_menu: 1,
                    name: 'Mock second name - 1',
                    position: 1,
                    url_path: 'urlpath/path.html',
                    children: [
                        {
                            children_count: '0',
                            id: 123,
                            include_in_menu: 1,
                            name: 'Mock third name - 1',
                            position: 1,
                            url_path: 'urlpath/path.html',
                            children: []
                        },
                        {
                            children_count: '0',
                            id: 124,
                            include_in_menu: 1,
                            name: 'Mock third name - 2',
                            position: 2,
                            url_path: 'urlpath/path.html',
                            children: []
                        }
                    ]
                },
                {
                    children_count: '0',
                    id: 12,
                    include_in_menu: 1,
                    name: 'Mock second name - 1',
                    position: 2,
                    url_path: 'urlpath/path.html',
                    children: []
                }
            ]
        }
    ]
};

export default mock;
