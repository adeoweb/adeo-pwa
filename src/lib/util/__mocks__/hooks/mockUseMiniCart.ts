import { TUseMiniCart } from 'src/peregrine/lib/talons/adeoweb/MiniCart/useMiniCart';
import { TPriceRange, TProductPrice } from 'src/lib/types/graphql/Product';

const productPrice: TProductPrice = {
    regular_price: {
        value: 10,
        currency: 'EUR'
    },
    discount: {
        amount_off: 0,
        percent_off: 0
    },
    final_price: {
        value: 0,
        currency: 'EUR'
    },
    fixed_product_taxes: [
        {
            amount: {
                value: 0,
                currency: 'EUR'
            },
            label: 'fresh air'
        }
    ]
};

const priceRange: TPriceRange = {
    maximum_price: productPrice,
    minimum_price: productPrice
};

const mock: TUseMiniCart = {
    cartItems: [
        {
            id: '123',
            product: {
                name: 'test product name',
                sku: 'testSku123',
                url_key: 'product.html',
                small_image: {
                    url: '',
                    label: ''
                },
                price_range: priceRange
            },
            quantity: 20
        }
    ],
    cartState: {
        addItemError: null,
        cartId: 'testCartId',
        details: {
            id: 123,
            is_virtual: false,
            items: [
                {
                    id: '123',
                    product: {
                        name: 'test product name',
                        sku: 'testSku123',
                        url_key: 'product.html',
                        small_image: {
                            url: '',
                            label: ''
                        },
                        price_range: priceRange
                    },
                    quantity: 20
                }
            ],
            prices: {
                grand_total: {
                    value: 999,
                    currency: 'EUR'
                }
            },
            shipping_addresses: [
                {
                    country: {
                        code: 'country code',
                        label: 'country label'
                    },
                    city: 'city',
                    firstname: 'firstname',
                    lastname: 'lastname',
                    street: ['street1', 'street2'],
                    telephone: 'telephone'
                }
            ]
        },
        detailsError: null,
        isLoading: false,
        isLoaded: false,
        isUpdatingItem: false,
        isAddingItem: false,
        removeItemError: null,
        updateItemError: null,
        isEmpty: false,
        derivedDetails: {
            currencyCode: 'EUR',
            numItems: 1,
            subtotal: 999
        }
    },
    currencyCode: 'EUR',
    handleBeginEditItem: () => {},
    handleDismiss: () => {},
    handleEndEditItem: () => {},
    handleClose: () => {},
    handleToggleCart: () => {},
    isEditingItem: false,
    isLoading: false,
    isMiniCartMaskOpen: false,
    isOpen: true,
    isUpdatingItem: false,
    numItems: 1,
    setStep: () => {},
    shouldShowFooter: true,
    step: '',
    subtotal: 999
};

export default mock;
