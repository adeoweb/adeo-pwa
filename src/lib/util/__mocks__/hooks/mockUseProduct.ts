import { TUseProduct } from 'src/peregrine/lib/talons/adeoweb/MiniCart/useProduct';

const mock: TUseProduct = {
    handleEditItem: () => {},
    handleFavoriteItem: () => {},
    handleRemoveItem: () => {},
    isFavorite: false,
    isLoading: false,
    productImage: '1bb5596fe10909d977e0c8185b18ce46/m/b/mb04-black-0.jpg',
    productName: 'testName',
    productOptions: [],
    productPrice: 10,
    productQuantity: 20,
    productUrlKey: 'product.html'
};

export default mock;
