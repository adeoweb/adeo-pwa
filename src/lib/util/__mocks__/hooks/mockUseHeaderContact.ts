import { TUseHeaderContact } from 'src/peregrine/lib/talons/adeoweb/Header/useHeaderContact';

const mock: TUseHeaderContact = {
    storePhone: '+000 1584 2578'
};

export default mock;
