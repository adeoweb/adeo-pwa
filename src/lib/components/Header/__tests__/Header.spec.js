import React from 'react';
import testRenderer from 'react-test-renderer';
import Header from '../Header';
import { BrowserRouter as Router } from 'react-router-dom';
import { IntlProvider } from 'react-intl';

import mockUseMenu from 'src/lib/util/__mocks__/hooks/mockUseMenu';
import mockUseProduct from 'src/lib/util/__mocks__/hooks/mockUseProduct';
import mockUseMiniCart from 'src/lib/util/__mocks__/hooks/mockUseMiniCart';
import mockUseLogo from 'src/lib/util/__mocks__/hooks/mockUseLogo';
import mockUseWelcomeMessage from 'src/lib/util/__mocks__/hooks/mockUseWelcomeMessage';
import mockUseHeaderContact from 'src/lib/util/__mocks__/hooks/mockUseHeaderContact';
import mockUseIsSignedIn from 'src/lib/util/__mocks__/hooks/mockUseIsSignedIn';
import mockUseSignOut from 'src/lib/util/__mocks__/hooks/mockUseSignOut';
import mockUseStoreList from 'src/lib/util/__mocks__/hooks/mockUseStoreList';
import mockUseProductCompare from 'src/lib/util/__mocks__/hooks/mockUseProductCompare';
import mockUseCompareDropdown from 'src/lib/util/__mocks__/hooks/mockUseCompareDropdown';

jest.mock('src/peregrine/lib/talons/adeoweb/Menu/useMenu', () => {
    return {
        useMenu: jest.fn(() => mockUseMenu)
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Logo/useLogo', () => {
    return {
        useLogo: () => mockUseLogo
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/IsSignedIn/useIsSignedIn', () => {
    return {
        useIsSignedIn: () => mockUseIsSignedIn
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/SignOut/useSignOut', () => {
    return {
        useSignOut: () => mockUseSignOut
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Header/useWelcomeMessage', () => {
    return {
        useWelcomeMessage: () => mockUseWelcomeMessage
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Header/useHeaderContact', () => {
    return {
        useHeaderContact: () => mockUseHeaderContact
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Header/useCartTrigger', () => {
    return {
        useCartTrigger: () => null
    };
});

jest.mock('src/peregrine/lib/context/adeoweb/app', () => {
    const state = {
        hasBeenOffline: false,
        isOnline: true,
        searchOpen: false
    };
    const api = {
        toggleSearch: jest.fn()
    };
    const useAppContext = jest.fn(() => [state, api]);

    return { useAppContext };
});

jest.mock('src/peregrine/lib/talons/adeoweb/MiniCart/useMiniCart', () => {
    return {
        useMiniCart: () => mockUseMiniCart
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/MiniCart/useProduct', () => {
    return {
        useProduct: () => mockUseProduct
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/App/useStoreList', () => {
    return {
        useStoreList: () => mockUseStoreList
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Product/useProductCompare', () => {
    return {
        useProductCompare: () => mockUseProductCompare
    };
});

jest.mock('src/peregrine/lib/talons/adeoweb/Product/useCompareDropdown', () => {
    return {
        useCompareDropdown: () => mockUseCompareDropdown
    };
});

test('Renders correctly', () => {
    const component = testRenderer.create(
        <Router>
            <IntlProvider locale="en">
                <Header />
            </IntlProvider>
        </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
});
