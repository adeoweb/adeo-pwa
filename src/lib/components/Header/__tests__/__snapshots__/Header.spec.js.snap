// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="header-top"
  >
    <div
      className="container"
    >
      <div
        className="header-left header-dropdowns"
      >
        <div
          className="header-dropdown"
        >
          <div>
            English
          </div>
          <div
            className="header-menu"
          >
            <ul>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="lt_LT"
                    src="test-file-stub"
                  />
                  Lithuanian
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="en_US"
                    src="test-file-stub"
                  />
                  English
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="ru_RU"
                    src="test-file-stub"
                  />
                  Russian
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="lv_LV"
                    src="test-file-stub"
                  />
                  Latvia
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="en_US"
                    src="test-file-stub"
                  />
                  English
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="ru_RU"
                    src="test-file-stub"
                  />
                  Russian
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="et_EE"
                    src="test-file-stub"
                  />
                  Estonians
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="en_US"
                    src="test-file-stub"
                  />
                  English
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <img
                    alt="ru_RU"
                    src="test-file-stub"
                  />
                  Russian
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="dropdown compare-dropdown active"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            className="dropdown-toggle"
            data-display="static"
            data-toggle="dropdown"
            role="button"
          >
            <i
              className="icon-retweet"
            />
            Compare (1)
          </div>
          <div
            className="dropdown-menu"
          >
            <div
              className="dropdownmenu-wrapper"
            >
              <ul
                className="compare-products"
              >
                <li
                  className="product"
                >
                  <button
                    className="btn-remove btn-link border-0 btn btn-primary"
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "background": "none",
                      }
                    }
                    title="Remove Product"
                    type="button"
                  >
                    <i
                      className="icon-cancel"
                    />
                  </button>
                  <h4
                    className="product-title"
                  >
                    <a
                      href="/product.html"
                      onClick={[Function]}
                    >
                      test product name
                    </a>
                  </h4>
                </li>
              </ul>
              <div
                className="compare-actions"
              >
                <button
                  className="action-link"
                  onClick={[Function]}
                >
                  Clear All
                </button>
                <a
                  className="btn btn-primary"
                  href="/catalog/product_compare/index"
                  onClick={[Function]}
                >
                  Compare
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="header-right"
      >
        <p
          className="welcome-msg"
        >
          Mock welcome message
           
        </p>
        <div
          className="header-dropdown dropdown-expanded"
        >
          <div
            className="links-button"
          >
            Links
          </div>
          <div
            className="header-menu"
          >
            <ul>
              <li>
                <a
                  href="/contactus"
                  onClick={[Function]}
                >
                  Contact
                </a>
              </li>
              <li>
                <a
                  href="/customer/login"
                  onClick={[Function]}
                >
                  Login
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="header-middle"
  >
    <div
      className="container"
    >
      <div
        className="header-left"
      >
        <a
          className="logo"
          href="/"
          onClick={[Function]}
        >
          <div
            className="undefined undefined"
          >
            <img
              alt="test"
              className="undefined undefined"
              height={100}
              loading="eager"
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
              style={
                Object {
                  "height": "100px",
                  "width": "100px",
                }
              }
              title="test"
              width={100}
            />
            <img
              alt="test"
              className="undefined undefined"
              height={100}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="test"
              style={
                Object {
                  "height": "100px",
                  "width": "100px",
                }
              }
              title="test"
              width={100}
            />
          </div>
        </a>
      </div>
      <div
        className="header-center"
      >
        <div>
          Loading...
        </div>
      </div>
      <div
        className="header-right"
      >
        <button
          className="mobile-menu-toggler"
          onClick={[Function]}
          type="button"
        >
          <i
            className="icon-menu"
          />
        </button>
        <div
          className="header-contact"
        >
          <span>
            Call us now
          </span>
          <a
            href="tel:+000 1584 2578"
          >
            <strong>
              +000 1584 2578
            </strong>
          </a>
        </div>
        <div
          className="dropdown cart-dropdown"
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="dropdown-toggle "
            data-display="static"
            data-toggle="dropdown"
            href="/cart"
            onClick={[Function]}
            role="button"
          >
            <span
              className="cart-count"
            >
              1
            </span>
          </a>
          <div
            className="dropdown-menu show"
          >
            <div
              className="dropdownmenu-wrapper"
            >
              <div
                className="dropdown-cart-header"
              >
                <span>
                  1
                   
                  Items
                </span>
                <a
                  href="/cart"
                  onClick={[Function]}
                >
                  View Cart
                </a>
              </div>
              <div
                className="dropdown-cart-products"
              >
                <div
                  className="product"
                >
                  <div
                    className="product-details"
                  >
                    <h4
                      className="product-title"
                    >
                      <a
                        href="/product.html"
                        onClick={[Function]}
                      >
                        testName
                      </a>
                    </h4>
                    <span
                      className="cart-product-info"
                    >
                      <span
                        className="cart-product-qty"
                      >
                        20
                      </span>
                      x 
                      <span>
                        €
                      </span>
                      <span>
                        10
                      </span>
                      <span>
                        .
                      </span>
                      <span>
                        00
                      </span>
                    </span>
                  </div>
                  <figure
                    className="product-image-container"
                  >
                    <a
                      className="product-image"
                      href="/product.html"
                      onClick={[Function]}
                    >
                      <div
                        className="undefined undefined"
                      >
                        <img
                          alt="testName"
                          className="undefined undefined"
                          height={67}
                          loading="eager"
                          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
                          style={
                            Object {
                              "height": "67px",
                              "width": "80px",
                            }
                          }
                          width={80}
                        />
                        <img
                          alt="testName"
                          className="undefined undefined"
                          height={67}
                          loading="lazy"
                          onError={[Function]}
                          onLoad={[Function]}
                          src="1bb5596fe10909d977e0c8185b18ce46/m/b/mb04-black-0.jpg"
                          style={
                            Object {
                              "height": "67px",
                              "width": "80px",
                            }
                          }
                          width={80}
                        />
                      </div>
                    </a>
                    <button
                      className="btn-remove btn btn-link"
                      onClick={[Function]}
                      title="Remove Product"
                    >
                      <i
                        className="icon-cancel"
                      />
                    </button>
                  </figure>
                </div>
              </div>
              <div
                className="dropdown-cart-total"
              >
                <span>
                  Total
                </span>
                <span
                  className="cart-total-price"
                >
                  <span>
                    €
                  </span>
                  <span>
                    999
                  </span>
                  <span>
                    .
                  </span>
                  <span>
                    00
                  </span>
                </span>
              </div>
              <div
                className="dropdown-cart-action"
              >
                <a
                  className="btn btn-block"
                  href="/checkout"
                  onClick={[Function]}
                >
                  Checkout
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sticky-wrapper"
  >
    <div
      className="header-bottom sticky-header"
    >
      <div
        className="container"
      >
        <a
          className="logo"
          href="/"
          onClick={[Function]}
        >
          <div
            className="undefined undefined"
          >
            <img
              alt="test"
              className="undefined undefined"
              height={100}
              loading="eager"
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
              style={
                Object {
                  "height": "100px",
                  "width": "100px",
                }
              }
              title="test"
              width={100}
            />
            <img
              alt="test"
              className="undefined undefined"
              height={100}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="test"
              style={
                Object {
                  "height": "100px",
                  "width": "100px",
                }
              }
              title="test"
              width={100}
            />
          </div>
        </a>
        <div
          className="dropdown cart-dropdown"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="dropdown-toggle"
            data-display="static"
            data-toggle="dropdown"
            href="/"
            role="button"
          >
            <span
              className="cart-count"
            >
              2
            </span>
          </a>
          <div
            className="dropdown-menu"
          >
            <div
              className="dropdownmenu-wrapper"
            >
              <div
                className="dropdown-cart-header"
              >
                <span>
                  2 Items
                </span>
                <a
                  href="cart.html"
                >
                  View Cart
                </a>
              </div>
              <div
                className="dropdown-cart-products"
              >
                <div
                  className="product"
                >
                  <div
                    className="product-details"
                  >
                    <h4
                      className="product-title"
                    >
                      <a
                        href="product.html"
                      >
                        Woman Ring
                      </a>
                    </h4>
                    <span
                      className="cart-product-info"
                    >
                      <span
                        className="cart-product-qty"
                      >
                        1
                      </span>
                      x $99.00
                    </span>
                  </div>
                  <figure
                    className="product-image-container"
                  >
                    <a
                      className="product-image"
                      href="product.html"
                    >
                      <img
                        alt="product"
                        src="src/lib/assets/images/products/cart/product-1.jpg"
                      />
                    </a>
                    <a
                      className="btn-remove"
                      href="/"
                      title="Remove Product"
                    >
                      <i
                        className="icon-cancel"
                      />
                    </a>
                  </figure>
                </div>
                <div
                  className="product"
                >
                  <div
                    className="product-details"
                  >
                    <h4
                      className="product-title"
                    >
                      <a
                        href="product.html"
                      >
                        Woman Necklace
                      </a>
                    </h4>
                    <span
                      className="cart-product-info"
                    >
                      <span
                        className="cart-product-qty"
                      >
                        1
                      </span>
                      x $35.00
                    </span>
                  </div>
                  <figure
                    className="product-image-container"
                  >
                    <a
                      className="product-image"
                      href="product.html"
                    >
                      <img
                        alt="product"
                        src="src/lib/assets/images/products/cart/product-2.jpg"
                      />
                    </a>
                    <a
                      className="btn-remove"
                      href="/"
                      title="Remove Product"
                    >
                      <i
                        className="icon-cancel"
                      />
                    </a>
                  </figure>
                </div>
              </div>
              <div
                className="dropdown-cart-total"
              >
                <span>
                  Total
                </span>
                <span
                  className="cart-total-price"
                >
                  $134.00
                </span>
              </div>
              <div
                className="dropdown-cart-action"
              >
                <a
                  className="btn btn-block"
                  href="checkout-shipping.html"
                >
                  Checkout
                </a>
              </div>
            </div>
          </div>
        </div>
        <nav
          className="main-nav"
        >
          <nav
            className="main-nav"
          >
            <ul
              className="menu sf-arrows sf-js-enabled"
              style={
                Object {
                  "touchAction": "pan-y",
                }
              }
            >
              <li
                className="megamenu-container "
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <a
                  className="sf-with-ul"
                  href="/mockurl.html"
                  onClick={[Function]}
                >
                  Mock name
                </a>
                <div
                  className="megamenu col"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  <div
                    className="row row"
                  >
                    <div
                      className="col-lg-8"
                    >
                      <div
                        className="row"
                      >
                        <div
                          className="col-lg-4"
                        >
                          <div
                            className="menu-title"
                          >
                            <a
                              href="/urlpath/path.html"
                              onClick={[Function]}
                            >
                              Mock second name - 1
                            </a>
                          </div>
                          <ul>
                            <li>
                              <a
                                href="/urlpath/path.html"
                                onClick={[Function]}
                              >
                                Mock third name - 1
                              </a>
                            </li>
                            <li>
                              <a
                                href="/urlpath/path.html"
                                onClick={[Function]}
                              >
                                Mock third name - 2
                              </a>
                            </li>
                          </ul>
                        </div>
                        <div
                          className="col-lg-4"
                        >
                          <div
                            className="menu-title"
                          >
                            <a
                              href="/urlpath/path.html"
                              onClick={[Function]}
                            >
                              Mock second name - 1
                            </a>
                          </div>
                          <ul />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
            </ul>
          </nav>
        </nav>
      </div>
    </div>
  </div>
</header>
`;
